pkg/tsdb/cloudwatch/query_transformer.go

```diff
@@ -104,10 +99,119 @@ func (e *cloudWatchExecutor) transformQueryResponsesToQueryResult(cloudwatchResp
 			queryResult.ErrorString = "Cloudwatch GetMetricData error: Too many datapoints requested - your search has been limited. Please try to reduce the time range"
 		}
 
+		eq, err := json.Marshal(executedQueries)
+		if err != nil {
+			plog.Error("Could not marshal executedString struct", err)
```

Comment:
Shouldn't you return if it fails here? I don't think you should be using `eq` further down, if the encoding failed.

Reply:
This is just meta data, so we don't need to return an error. But we should definitely add key/value pair to plog!

Reply:
Or actually I think you're right. Lets return an error

Reply:
I actually didn't mean to return an error, but to return in the first place. This code just continues in case of an error.